feature/settings/src/main/java/com/google/jetpackcamera/settings/SettingsViewModel.kt

```diff
@@ -49,9 +52,9 @@ private val fpsOptions = setOf(FPS_15, FPS_30, FPS_60)
 @HiltViewModel
 class SettingsViewModel @Inject constructor(
     private val settingsRepository: SettingsRepository,
-    private val permissionChecker: PermissionChecker,
     constraintsRepository: ConstraintsRepository
 ) : ViewModel() {
+    private var grantedPermissions = mutableSetOf<String>()
```

Comment:
You should make this a `MutableStateFlow<Set<String>>` so that it can be combined into the below flow. Otherwise you have a race between this being set and the settings/constraints being updated.